src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -238,7 +238,8 @@ void primitive_inst::realloc_if_needed() {
     GPU_DEBUG_GET_INSTANCE(debug_config);
     GPU_DEBUG_PROFILED_STAGE(instrumentation::pipeline_stage::memory_allocation);
 
-    auto actual_layout = _impl_params->output_layout;
+    auto fake_params = _node->type()->get_fake_aligned_params(*_impl_params);
+    auto actual_layout = fake_params.output_layout;
     OPENVINO_ASSERT(actual_layout.is_static(), "[GPU] Can't realloc mem for dynamic layout");
```

Comment:
[random spot] Do we have a unit test that covers this feature? If no, please add it.

Reply:
Added